src/index/index.rs

```diff
@@ -521,30 +521,24 @@ impl Index {
         load_metas(self.directory(), &self.inventory)
     }
 
-    /// Open a new index writer. Attempts to acquire a lockfile.
+    /// Open a new index writer with the given options. Attempts to acquire a lockfile.
     ///
     /// The lockfile should be deleted on drop, but it is possible
     /// that due to a panic or other error, a stale lockfile will be
     /// left in the index directory. If you are sure that no other
     /// `IndexWriter` on the system is accessing the index directory,
     /// it is safe to manually delete the lockfile.
     ///
-    /// - `num_threads` defines the number of indexing workers that should work at the same time.
-    ///
-    /// - `overall_memory_budget_in_bytes` sets the amount of memory allocated for all indexing
-    ///   thread.
-    ///
-    /// Each thread will receive a budget of `overall_memory_budget_in_bytes / num_threads`.
+    /// - `options` defines the writer configuration which includes things like buffer sizes,
+    ///   indexer threads, etc...
     ///
     /// # Errors
-    /// If the lockfile already exists, returns `Error::DirectoryLockBusy` or an `Error::IoError`.
+    /// If the lockfile already exists, returns `TantivyError::LockFailure`.
     /// If the memory arena per thread is too small or too big, returns
     /// `TantivyError::InvalidArgument`
-    pub fn writer_with_num_threads_and_num_merge_threads<D: Document>(
```

Comment:
is this method published in a version? If so can we keep an implem of it and call writer_with_options?

Reply:
IIRC it isn't, this method was added in https://github.com/quickwit-oss/tantivy/pull/2535 ~3 weeks ago, I think the last published version was April. Although I think @Barre is probably using this already (pinging so they're aware of this change.)

Reply:
that's ok then. As long as it isn't published!